File: 1177. Can Make Palindrome from Substring.py
class Solution:
    def canMakePaliQueries(self, s: str, queries: List[List[int]]) -> List[bool]:
        c = [[0]*26]
        c[0][ord(s[0])-97] += 1
        for i in range(1, len(s)):
            nxt = c[-1][:]
            nxt[ord(s[i])-97] += 1
            c.append(nxt)
        ans = [k>=sum([(c[r][i]-(0 if l==0 else c[l-1][i])) & 1 for i in range(26)])//2 
               for l,r,k in queries]
        return ans

File: 1177. Can Make Palindrome from Substring.cpp
class Solution {
public:
    vector<bool> canMakePaliQueries(string s, vector<vector<int>>& queries) {
        int n = s.size();
        vector<vector<int>> c(1, vector<int>(26));
        ++c[0][s[0]-'a'];
        for (int i=1; i<n; ++i) {
            c.push_back(c.back());
            ++c.back()[s[i]-'a'];
        }
        vector<bool> ans;
        for (auto& q : queries) {
            vector<int> a(26);
            for (int i=0; i<26; ++i) 
                a[i] = c[q[1]][i] - (q[0]==0 ? 0 : c[q[0]-1][i]);
            int k = q[2], od = 0;
            for (int i=0; i<26; ++i)
                if (a[i] & 1)
                    ++od;
            ans.push_back(k >= od/2);
        }
        return ans;
    }
};
